crud.controller.js

```diff
@@ -585,7 +601,17 @@ CrudController.prototype = {
             promise = self.model.find({ "_id": { "$in": docIds } });
         }
         return promise
-            .then(documents => {
+            .then(async documents => {
+                if (restrictOnError) {
+                    var startSession = await isTransactionSupported(self, res);
+                    if (startSession) {
+                        req.session = session = await self.model.startSession();
```

Comment:
We should define the [TransactionOptionsObject](https://mongodb.github.io/node-mongodb-native/3.6/api/global.html#TransactionOptions) for a session. This session that we are operating should define the _readPreference_, _writeConcern_ and _readConcern_.
It should be 
```
readPreference: Primary
write and readConcern: majority
```